console/frontend/src/main/frontend/src/app/views/security-items/security-items.service.ts

```diff
@@ -81,16 +163,28 @@ export type Links = { links: Link[] };
   providedIn: 'root',
 })
 export class SecurityItemsService {
-  constructor(
-    private http: HttpClient,
-    private appService: AppService,
-  ) {}
+  private securityItemsCache: SecurityItems | null = null;
+  private endpointsWithRolesCache: Links | null = null;
+  private readonly http: HttpClient = inject(HttpClient);
+  private readonly appService: AppService = inject(AppService);
 
   getSecurityItems(): Observable<SecurityItems> {
-    return this.http.get<SecurityItems>(`${this.appService.absoluteApiPath}securityitems`);
+    if (this.securityItemsCache) {
+      return of(this.securityItemsCache);
+    }
```

Comment:
Maybe this should have a way to be cleared or updated too, for example when a user logs out

Reply:
Good point. I also thought to tie this into the auth service.

Reply:
Added a clear to the logout. This will also trigger if redirected to logout page.